Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -193,15 +193,13 @@ std::optional<InteractionRegion> interactionRegionForRenderedRegion(RenderObject
     if (!matchedElement)
         return std::nullopt;
 
-    if (auto* linkElement = matchedElement->enclosingLinkEventParentOrSelf())
-        matchedElement = linkElement;
     if (auto* buttonElement = ancestorsOfType<HTMLButtonElement>(*matchedElement).first())
         matchedElement = buttonElement;
-
```

Comment:
Would it be behaviorally any better to walk up the tree once instead of for each? (it's not worth it for the tiny perf win, but might be worth it if it makes choosing the right ancestor better in any way).

Reply:
could make future tweaks clearer too... taking a look!